src/platform/mt8186/lib/clk.c

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Mediatek. All rights reserved.
+//
+// Author: Allen-KH Cheng <allen-kh.cheng@mediatek.com>
+//         Tinghan Shen <tinghan.shen@mediatek.com>
+
+#include <platform/drivers/mt_reg_base.h>
+#include <sof/common.h>
+#include <sof/lib/clk.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/uuid.h>
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+#include <sof/trace/trace.h>
+
+DECLARE_SOF_UUID("clkdrv", clkdrv_uuid, 0x23b12fd5, 0xc2a9, 0x41a8,
```

Comment:
Need to generate another unique uuid for mt8186, don't use the same one with mt8195.